larsim/EventGenerator/POTaccumulator_module.cc

```diff
@@ -0,0 +1,307 @@
+/**
+ * @file   larsim/EventGenerator/POTaccumulator_module.cc
+ * @brief  Module summing all POT from the input files.
+ * @date   April 29, 2020
+ * @author Gianluca Petrillo (petrillo@slac.stanford.edu)
+ */
+
+// LArSoft libraries
+#include "larcorealg/CoreUtils/get_elements.h"
+#include "larcoreobj/SummaryData/POTSummary.h"
+
+// framework libraries
+#include "art/Framework/Core/EDAnalyzer.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/SubRun.h"
+#include "art/Framework/Principal/Handle.h"
+#include "canvas/Utilities/InputTag.h"
+#include "fhiclcpp/types/Atom.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+// C/C++ standard libraries
+#include <map>
+#include <string>
+#include <algorithm> // std::for_each()
+
+
+// -----------------------------------------------------------------------------
+namespace sim { class POTaccumulator; }
+
+/**
+ * @brief  Prints on console the total Protons On Target from the input subruns.
+ *
+ * This module collects information from each of the subrun in the input files
+ * and prints the total of the protons on target. It also optionally prints the
+ * total for each observed run.
+ * If a subrun is met more than once, the information from all subrun fragments
+ * are added together (i.e. it is assumed that summary information is
+ * complementary rather than duplicate).
```

Comment:
How familiar are you with the framework's automatic product aggregation handling for (sub)runs?

https://cdcvs.fnal.gov/redmine/projects/art/wiki/Run_and_SubRun_products

The framework does careful bookkeeping to ensure that a (sub)run product presented to the user represents either an entire (sub)run or a well-formed range of validity.  Unless users are inserting `POTsummary` products in their files with different semantics (i.e. `full(Sub)Run` vs. `(sub)RunFragment`), the `POTsummary` objects will be complementary/disjoint.

Your code below does not check for that explicitly, even though it could.  Let me know if you're interested in how to do it.